okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -500,7 +501,11 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
         writer.windowUpdate(0, (windowSize - DEFAULT_INITIAL_WINDOW_SIZE).toLong())
       }
     }
-    Thread(readerRunnable, connectionName).start() // Not a daemon thread.
+    // Thread doesn't use client Dispatcher, since it is scoped potentially across clients via
+    // ConnectionPool.
+    Thread(readerRunnable, connectionName).apply {
```

Comment:
An alternative design here would be to add TaskRunner.NON_DAEMON.  and then use the boolean to choose between INSTANCE and NON_DAEMON.

This would add automatic tracking of these threads, so we can check everything is shutdown in OkHttpClientTestRule also.

Worth doing?